core/server/services/rss/generate-feed.js

```diff
@@ -87,17 +87,18 @@ generateFeed = function generateFeed(baseUrl, data) {
             }
         });
 
-    data.posts.forEach(function forEach(post) {
-        var item = generateItem(post, siteUrl, data.secure);
-
-        filters.doFilter('rss.item', item, post).then(function then(item) {
-            feed.item(item);
+    return Promise.all(
+        data.posts.map(function map(post) {
```

Comment:
First of all: you found a logic bug 👍 

The solution you have suggested can probably cause trouble. I haven't tested it, but i think:

The posts (`data.posts`) will be provided in a specific order. You are making use of `Array.prototype.map`, which will simply iterate over all posts, without waiting for the promise result of the filter for the post item (`ilters.doFilter('rss.item'`). If a filter takes longer for an item, you will add the rss items in wrong order.

Reply:
Of course, you're right, Katharina. Although, it's not about the solution, but the initial (probably intentional) logic I tried to preserve ) I'll fix it too, thanks for the feedback!